### PR TITLE
Terser casper tests

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

Commit log & the diffstat say it all.
Adding this many reviewers in hopes of a swift merge :)

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
For your convenience, here's publishBlock:
https://github.com/rchain/rchain/blob/b6c6ceeead8769d2a5f8cd3e211b5af6e4387e53/casper/src/test/scala/coop/rchain/casper/helper/TestNode.scala#L110-L114

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala [hunk 1]
```diff
@@ -40,14 +40,10 @@ class MultiParentCasperMergeSpec
         )
         block0 <- nodes(0).addBlock(deploys(0))
         block1 <- nodes(1).addBlock(deploys(1))
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
+        _      <- nodes.toList.traverse_(_.receive())
```
Comment (itegulov) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala:43]: The new code is not equivalent to the old one but looking at the test I guess 4 recieves are redundant anyway.

**Review by itegulov — APPROVED** [review]

**Review by tymm — APPROVED** [review]

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala (hunk 1 above)
Comment (marcin-rzeznicki) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala:43]: Sure `.replicateA(2)`

**Review by marcin-rzeznicki — APPROVED** [review]

**Comment by marcin-rzeznicki** [comment]
`nodes.toList.filter(_ != this).traverse_(_.receive()) `

Hmm, `traverseFilter` should be a tad more efficient  - it does not matter here but it avoids allocations of intermediate collections

**Pull request merged**